tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -480,13 +480,29 @@ def initKubernetesClientRunner(self) -> KubernetesClientRunner:
 
     def tearDown(self):
         logger.info('----- TestMethod %s teardown -----', self.id())
+        # Todo(xuanwn): Add same check to XdsUrlMapTestCase.
+        logger.debug('Getting pods restart times')
+        client_restarts: int = self.client_runner.get_pod_restarts(
+            self.client_runner.deployment)
+        server_restarts: int = self.server_runner.get_pod_restarts(
+            self.server_runner.deployment)
         retryer = retryers.constant_retryer(wait_fixed=_timedelta(seconds=10),
                                             attempts=3,
                                             log_level=logging.INFO)
         try:
             retryer(self.cleanup)
         except retryers.RetryError:
             logger.exception('Got error during teardown')
+        finally:
+            # We should fail test if pod restarted during test (b/269192257).
+            self.assertEqual(client_restarts,
+                             0,
+                             msg=('Client pods unexpectedly restarted'
```

Comment:
Thanks for the detailed message here! Very useful for debugging.
Let's also add a note suggesting what a pod restart means:

`In most cases, this is caused by the test client app crash.`

and 

`In most cases, this is caused by the test server app crash.`